tests/scenarios/compat-stage2-test.ts

```diff
@@ -491,134 +577,214 @@ stage2Scenarios
-          .resolves('#embroider_compat/components/second-choice')
+          .resolves(/\/components\/second-choice.hbs\/-embroider-pair-component/)
           .toModule()
-          .isTemplateOnlyComponent('./templates/components/second-choice.hbs', 'rule-driven mustache string literal');
+          .withContents(contents => {
+            assert.ok(
+              /setComponentTemplate\(template, templateOnlyComponent\(undefined, "second-choice"\)\);/.test(contents)
+            );
+            return true;
+          });
       });
 
-      test('curly.hbs', function () {
+      test('curly.hbs', function (assert) {
         let expectModule = expectAudit.module('./templates/curly.hbs');
         expectModule
-          .resolves('#embroider_compat/ambiguous/hello-world')
-          .to('./node_modules/my-addon/_app_/components/hello-world.js', 'explicit dependency');
+          .resolves(/my-addon\/_app_\/components\/hello-world/)
+          .toModule()
+          .codeContains('');
+
         expectModule
-          .resolves('#embroider_compat/components/third-choice')
+          .resolves(/\/components\/third-choice.hbs\/-embroider-pair-component/)
           .toModule()
-          .isTemplateOnlyComponent('./templates/components/third-choice.hbs');
+          .withContents(contents => {
+            assert.ok(
+              /setComponentTemplate\(template, templateOnlyComponent\(undefined, "third-choice"\)\);/.test(contents)
+            );
+            return true;
+          });
+
         expectModule
-          .resolves('#embroider_compat/components/first-choice')
+          .resolves(/\/components\/first-choice.hbs\/-embroider-pair-component/)
           .toModule()
-          .isTemplateOnlyComponent('./templates/components/first-choice.hbs');
+          .withContents(contents => {
+            assert.ok(
+              /setComponentTemplate\(template, templateOnlyComponent\(undefined, "first-choice"\)\);/.test(contents)
+            );
+            return true;
+          });
       });
 
-      test('addon/hello-world.js', function () {
-        expectAudit.module('./node_modules/my-addon/components/hello-world.js').codeEquals(`
-        window.define("my-app/components/second-choice", function () {
-          return importSync("#embroider_compat/components/second-choice");
-        });
-        window.define("my-addon/synthetic-import-1", function () {
-          return importSync("../synthetic-import-1");
-        });
-        import Component from '@ember/component';
-        import layout from '../templates/components/hello-world';
-        import computed from '@ember/object/computed';
-        import somethingExternal from 'not-a-resolvable-package';
-        import { importSync } from "@embroider/macros";
-        export default Component.extend({
-          dynamicComponentName: computed('useDynamic', function () {
-            return this.useDynamic || 'default-dynamic';
-          }),
-          layout
-        });
```

Comment:
Do we want to keep asserting all the imports are present, or do we just want enough proof that this is the right file that has been loaded?